pilot/proxy/envoy/route.go

```diff
@@ -228,9 +228,8 @@ func buildHTTPRouteV1(config model.Config, service *model.Service, port *model.P
 		}
 	} else {
 		// default route for the destination
-		cluster := buildOutboundCluster(destination, port, nil)
-		route.Cluster = cluster.Name
-		route.clusters = append(route.clusters, cluster)
+		route.Cluster = defaultCluster.Name
+		route.clusters = append(route.clusters, defaultCluster)
```

Comment:
Thsi does not look right.. The abstraction seems broken if we compute clusters for weighted clusters but not for the non weighted case.

You can overwrite the computed cluster at the call site as i described above, unless there is a major restriction